src/common/pagesApi.js

```diff
@@ -1,7 +1,7 @@
 import { http } from "./http";
 
-export function fetchPages() {
-  return http.get("/uiapi/page");
+export function fetchPages(queryString) {
+  return queryString ? http.get(`/uiapi/page?${queryString}`) : http.get("/uiapi/page");
```

Comment:
nit-pick: This could be simplified using a [default parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)
```suggestion
export function fetchPages(queryString = "") {
  return http.get(`/uiapi/page?${queryString}`);
```

Reply:
It would make it simpler but the URL would always have the question mark present, is that ok?

Reply:
Never mind, forgot that this is just the request URL and isn't really client facing anyway.

Reply:
Suggestion taken.